lib/rspec/buildkite/analytics/tracer.rb

```diff
@@ -13,15 +13,15 @@ def initialize(section, start_at, end_at, detail)
         @children = []
       end
 
-      def as_json
+      def as_hash
         {
           section: section,
           start_at: start_at,
           end_at: end_at,
           duration: end_at - start_at,
           detail: detail,
-          children: children.map(&:as_json),
-        }
+          children: children.map(&:as_hash),
+        }.with_indifferent_access
```

Comment:
Why `with_indifferent_access`?

Reply:
Not sure exactly why this was happening, but the send queue data for a single result was a hash with symbol keys, but for all the results was a hash with string keys?? So [this line](https://github.com/buildkite/rspec-buildkite-analytics/pull/65/files#diff-e04810cfd3eb5dfc5d3661dbedcbf1ca7bfdc3d88e0c9a01d2ddab0fec126260R208) would fail as the ids couldn't always be accessed via the symbol attribute. Making the underlying structure a hash with indifferent access seemed like the easiest way of getting around it

Reply:
Active Support dependency got removed in https://github.com/buildkite/rspec-buildkite-analytics/pull/63 — So we need to do this in Ruby hmm

Reply:
#63 doesn't fully remove the AS dependency. It's still marked as a dep of the gem, and it uses activesupport notifications.

In #63 I was avoiding using AS methods though, because for some reason they weren't being added to `DateTime` in my non-rails app. Possibly because my app didn't require `date`?

Reply:
Ooops sorry I did not read the Pull Request in depth. We can still use `with_indifferent_access` 👍🏼

Reply:
> In #63 I was avoiding using AS methods though, because for some reason they weren't being added to `DateTime` in my non-rails app. Possibly because my app didn't require `date`?

[`DateTime.current`](https://api.rubyonrails.org/classes/DateTime.html#method-c-current) is a Rails extension so not available in non-Rails app.